### PR TITLE
Shopping UI review first.

### PR DESCRIPTION
**Pull request by irshad-alam**

### CODE REVIEW

File: app/styles/app.css [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  top: 0;
+}
+
+.header div{
+  text-align: right;
+}
+.header div a {
+  color: darkorange;
+  border: black thin solid;
+  background-color: #f1f1f1;
```
Comment (ruphaafd) [inline app/styles/app.css:17]: We can define all the color codes on top in  CSS variables and reuse the variable name across the file.

File: app/styles/app.css [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  bottom: 0;
+  width: 100%;
+  background-color: #68a5ff;
+  color: white;
+  text-align: center;
+}
+
+body {
+  font-family: "Lato", sans-serif;
+}
```
Comment (ruphaafd) [inline app/styles/app.css:34]: Body styles better to be placed at the top of css rules

File: app/authenticators/shoppin-auth.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      const data = JSON.stringify({
+        user_session: {
+          login: login,
+          password: password,
+          remember_me: rememberMe
+        }
+      });
+
+      return new Promise(function (resolve, reject) {
+        fetch('http://localhost:3000/api/v1/user/token',{
```
Comment (ruphaafd) [inline app/authenticators/shoppin-auth.js:17]: `http://localhost:3000/api/v1/user/token` - You can store this in a constant and reuse the variable.

File: app/authenticators/shoppin-auth.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+      const data = JSON.stringify({
+        user_session: {
+          login: login,
+          password: password,
+          remember_me: rememberMe
+        }
+      });
+
+      return new Promise(function (resolve, reject) {
```
Comment (ruphaafd) [inline app/authenticators/shoppin-auth.js:16]: Also, using Promise is a great approach, you can also try using async and await, to try varieties. Just a thought. This is a good Ember library to use asynchronous code - http://ember-concurrency.com/docs/introduction/ - Check it out

File: app/components/common/header.hbs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div class="header">
+  <h1>shoppin.com</h1>
+  <div>
+    <LinkTo @route="dashboard">Home</LinkTo>
+    {{#if this.session.isAuthenticated}}
+      <LinkTo @route="cart">Cart</LinkTo>
+      <a onclick={{action 'invalidate'}}>Logout({{this.session.data.authenticated.login}})</a>
```
Comment (ruphaafd) [inline app/components/common/header.hbs:7]: By default, the action that you specify to an element will get attached to the click event only, so it's suggested not to explicitly give `onclick`.

Comment (ruphaafd) [inline app/components/common/header.hbs:7]:
```suggestion
      <a {{action 'invalidate'}}>Logout({{this.session.data.authenticated.login}})</a>
```

Comment (ruphaafd) [inline app/components/common/header.hbs:7]: A small note: even without explicitly giving `this` you can access the session directly in the hbs file.
Like this `{{session.data.authenticated.login}}`

File: app/components/common/header.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import { action } from "@ember/object";
+
+export default class CommonHeaderComponent extends Component {
+
+  @service
+  session
+
+  @action
+  invalidate() {
+    alert("Logging out");
```
Comment (ruphaafd) [inline app/components/common/header.js:12]: Can you please remove the alerts?

File: app/components/display.hbs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<div class="sidenav">
+  {{#each this.categories as |category|}}
+    <a {{on 'click' (fn this.changeItem category) }}>
```
Comment (ruphaafd) [inline app/components/display.hbs:3]: What are we trying to achieve here?

File: app/authenticators/shoppin-auth.js (hunk 3 above)
Comment (irshad-alam) [reply app/authenticators/shoppin-auth.js]: Done. Moved to env variable.

File: app/authenticators/shoppin-auth.js (hunk 4 above)
Comment (irshad-alam) [reply app/authenticators/shoppin-auth.js:17]: Sure! Will go through it too. Thank you.

**Pull request merged**